components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -348,11 +422,32 @@ std::string ExtractPdfContent(const ui::AXNode* pdf_root) {
   return pdf_content;
 }
 
+// Obtains a patch URL from a pull request URL
+std::optional<GURL> GetGithubPatchURLForPRURL(const GURL& url) {
+  if (!url.is_valid() || !url.SchemeIsCryptographic() ||
+      url.host() != "github.com") {
+    return std::nullopt;
+  }
+
+  std::vector<std::string> path_parts = base::SplitString(
+      url.path(), "/", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  // Only handle URLs of the form: /<user>/<repo>/pull/<number>
+  if (path_parts.size() < 4 || path_parts[2] != "pull") {
+    return std::nullopt;
+  }
+  std::string patch_url_str = url.GetWithEmptyPath().spec() + path_parts[0] +
+                              "/" + path_parts[1] + "/pull/" + path_parts[3] +
```

Comment:
Isn't `/pull/` already in `path_parts[2]`? Also please use `base::JoinString`.

Reply:
More specifically `pull` but I agree `base::JoinString` will be a good cleanup here.